src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -538,16 +542,14 @@ static class SystemModulesClassGenerator {
         private static final MethodTypeDesc MTD_Map = MethodTypeDesc.of(CD_Map);
         private static final MethodTypeDesc MTD_MapEntry_Object_Object = MethodTypeDesc.of(CD_Map_Entry, CD_Object, CD_Object);
         private static final MethodTypeDesc MTD_Map_MapEntryArray = MethodTypeDesc.of(CD_Map, CD_Map_Entry.arrayType());
-        private static final MethodTypeDesc MTD_Set_ObjectArray = MethodTypeDesc.of(CD_Set, CD_Object.arrayType());
+
+        static final int PAGING_THRESHOLD = 512; // An arbitrary number as this likely generate minimum ~4K code
```

Comment:
Can this be a default paging threshold value for `CollectionSnippetBuilder` and override it when it's necessary?